File: Assets/AdvancedInputField/Scripts/NativeKeyboard/INativeKeyboardCallback.cs
namespace AdvancedInputFieldPlugin
{
	public interface INativeKeyboardCallback
	{
		void OnTextEditUpdate(string text, int selectionStartPosition, int selectionEndPosition);
		void OnKeyboardShow();
		void OnKeyboardHide();
		void OnKeyboardDone();
		void OnKeyboardNext();
		void OnKeyboardCancel();
		void OnKeyboardHeightChanged(int height);
		void OnHardwareKeyboardChanged(bool connected);
	}
}

File: Android/NativeKeyboard/NativeKeyboard/src/main/java/com/jeroenvanpienbroek/nativekeyboard/INativeKeyboardCallback.java
package com.jeroenvanpienbroek.nativekeyboard;

public interface INativeKeyboardCallback
{
    public void OnTextEditUpdate(String text, int selectionStartPosition, int selectionEndPosition);
    public void OnAutofillUpdate(String text, int autofillType);
    public void OnKeyboardShow();
    public void OnKeyboardHide();
    public void OnKeyboardDone();
    public void OnKeyboardNext();
    public void OnKeyboardCancel();
    public void OnSpecialKeyPressed(int specialKeyCode);
    public void OnKeyboardHeightChanged(int height);
    public void OnHardwareKeyboardChanged(boolean connected);
}
